STYLING.md

```diff
@@ -0,0 +1,225 @@
+        ]}
+    >
+        {props.children}
+    </Text>
+);
+```
+
+On the other hand, if we are copying and pasting some chunks of JSX from one place to another then that might be a sign that we need either a new component or a new reusable style.
+
+## Use the "Rule of Three"
+
+Any array of styles associated with a single type of component or element that has at least 3 identical usages should be refactored into either:
+
+- A new component with that same array of helper styles
+
+- A new style that is a composite of all helper styles
+
+## Inline Styles
+
+**Inline styles are forbidden.** If we run into a case where we feel it's necessary to conditionally render some styles. There are two options:
+
+1. Create a helper function in `styles.js` and pass any modifying parameters to that function.
+1. Conditionally render a style object inline by defaulting to `undefined`.
+
+```jsx
+// Bad - Do not use inline styles
+const TextWithPadding = props => (
+    <Text style={{
+        padding: 10,
+        whiteSpace: props.shouldWrap ? 'wrap' : 'nowrap',
+    }}>
+        {props.children}
+    </Text>
+);
+
+// Good
+const TextWithPadding = props => (
+    <Text
+        style={[
+            styles.p5,
+            props.shouldWrap ? undefined : styles.noWrap,
+        ]}
+    >
+        {props.children}
+    </Text>
+);
+
+// Good
+const TextWithPadding = props => (
+    <Text
+        style={[
+            styles.p5,
+            getTextWrapStyle(props.shouldWrap)
+        ]}
+    >
+        {props.children}
+    </Text>
+);
+```
+
+## Don't Go Style Fishing
+
+There are many styles in the `styles.js` file. It is generally a bad practice to grab style meant for a specific use case and utilize it for some other use case without changing it's name to make it more general. If we think we see a style that might be appropriate for reuse, but does not have a generic name then we should rename it instead of using it directly.
+
+```jsx
+// Bad - Reuses style without generalizing style name
+const SettingsScreen = props => (
+    <View>
+        <Text style={[styles.settingsScreenText]}>
+            Expensify
+        </Text>
+    </View>
+);
+
+const SomeOtherScreen = props => (
+    <View>
+        <Text style={[styles.settingsScreenText]}>
+            Expensify.cash
+        </Text>
+    </View>
+);
+
+// Good
+const SettingsScreen = props => (
+    <View>
+        <Text style={[styles.defaultScreenText]}>
+            Expensify
+        </Text>
+    </View>
+);
+
+const SomeOtherScreen = props => (
+    <View>
+        <Text style={[styles.defaultScreenText]}>
+            Expensify.cash
+        </Text>
+    </View>
+);
+```
+
+## When and How to Pass Styles via Props
+
+In some cases, we may want a more complex component to allow a parent to modify a style of one of it's child elements. In other cases, we may have a very simple component that has one child which has a `style` prop. Let's look at how to handle these two examples.
```

Comment:
This is all stemming from this conversation -> https://github.com/Expensify/Expensify.cash/pull/1878#discussion_r605174892

I'm cool with whatever we decide to do as long as it's consistent

Reply:
Yeah, as stated in that other thread, I think that having the extra handling where a style can be an object or an array will be simpler than actually tying it to the plurality of the prop name. I have three reasons:

1) Built-in React Native components such as `View` and `Pressable` can take either and object or array of styles, so having our components follow the same pattern is a good thing for consistency.
2) React native provides a pretty easy way to deal with styles that can be an array or an object in `StyleSheet.flatten`, and I think that's a pretty easy pattern to pick up on and remember. "If you have a style prop, flatten it!" The docs make me think there might be some performance implications of overusing that though, so maybe the tertiary pattern is better 🤷🏼‍♂️
3) When dealing with contributors (esp. those for whom English is a second language) and reviewing PRs, I just have a feeling that this `style` == `Object` and `styles` == `Array` might slip through the cracks and end up not being thoroughly enforced.

Reply:
Nice thoughts. 

I guess my concern with making all components behave like React Native components is that we'll always have to use a bunch of `StyleSheet.flatten()` everywhere and it makes adding styles to existing default styles kind of tricky. 

Maybe if we are going to standardize on something we can just do "always use an array"?

If extra style props are always arrays then we can spread them or pass them directly and don't have to handle different types.

Reply:
I like the idea of always standardizing on arrays. I think that solves the problem pretty nicely and makes it very clear how something should be done.

Reply:
Always using arrays is fine with me.